core/src/main/java/hudson/model/Descriptor.java

```diff
@@ -806,8 +808,31 @@ public boolean configure( StaplerRequest req ) throws FormException {
      *      to keep the client in the same config page.
      */
     public boolean configure( StaplerRequest req, JSONObject json ) throws FormException {
-        // compatibility
-        return configure(req);
+
+        if (Util.isOverridden(Descriptor.class, getClass(), "configure", StaplerRequest.class)) {
+            // compatibility
+            return configure(req);
+        }
+
+        try (BulkChange bc = new BulkChange(this)) {
+            req.bindJSON(this, json);
+            bc.commit();
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, "Failed to save "+getConfigFile(),e);
+        }
+        return true;
+    }
+
+
+    /**
+     * Configure descriptor instance form structured form submission where this specific descriptor is nested in
+     * a larger payload, which typically happens when Descriptor doesn't have a dedicated configuration view.
+     */
```

Comment:
Add `@since TODO` if you want it to be a part of API. Otherwise makes sense to restrict it

Reply:
Change to consider 👍 as for all the new public methods in this PR